### PR TITLE
Fix part of #44: Hi-fi UI admin auth screen in add profile

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
## Explanation

Mock Link:
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/a841330e-efe5-4bdb-acdd-4d6e52a59571/PC-MP-Admin-Authorization-Empty-

small

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt [hunk 1]
```diff
@@ -24,11 +24,14 @@ class AdminAuthActivityPresenter @Inject constructor(
 
   /** Binds ViewModel and sets up text and button listeners. */
   fun handleOnCreate() {
-    activity.title = activity.getString(R.string.add_profile_title)
-    activity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
-    activity.supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_close_white_24dp)
-
     val binding = DataBindingUtil.setContentView<AdminAuthActivityBinding>(activity, R.layout.admin_auth_activity)
+
+    binding.adminAuthToolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt]: Add this icon directly in xml file:
`app:navigationIcon="@drawable/ic_close_white_24dp"`

File: app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt [hunk 2]
```diff
@@ -24,11 +24,14 @@ class AdminAuthActivityPresenter @Inject constructor(
...
+    binding.adminAuthToolbar.setNavigationOnClickListener {
+      (activity as AdminAuthActivity).finish()
+    }
+    binding.adminAuthToolbar.title = activity.getString(R.string.add_profile_title)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt]: Also, this is not needed, xml file already has this attribute

File: app/src/main/java/org/oppia/app/profile/AdminAuthViewModel.kt [hunk 3]
```diff
@@ -9,4 +12,21 @@ import javax.inject.Inject
...
+  var isSubmitButtonActive = ObservableField<Boolean>(false)
+
+  @Bindable
+  fun getInputPinTextWatcher(): TextWatcher {
+    return object : TextWatcher {
+      override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
+      }
+
+      override fun onTextChanged(answer: CharSequence, start: Int, before: Int, count: Int) {
+        isSubmitButtonActive.set(answer.toString().trim().length==5)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/profile/AdminAuthViewModel.kt]: Nit spacing: isSubmitButtonActive.set(answer.toString().trim().length == 5)

**Review by rt4914 — CHANGES_REQUESTED** [review]
Screenshot_1578042366

Submit button should be visible above keyboard, when keyboard is displayed.
`DONE` / `TICK` in keyboard should submit the pin automatically.

Also, we need to add test cases too, like:

1. Submit button is displayed by default
2. Submit button is active only when 5 characters are entered.

**Comment by rt4914** [comment]
@nikitamarysolomanpvt You have assigned this PR to @mschanteltc so do you have any specific doubt regarding this mocks. If yes, then mention it and if no, then unassign @mschanteltc .

**Review by veena14cs — APPROVED** [review]
LGTM. Please address other comments.

File: app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt]: Done.

File: app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt (hunk 2 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt]: Removed.

File: app/src/main/java/org/oppia/app/profile/AdminAuthViewModel.kt (hunk 3 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/profile/AdminAuthViewModel.kt]: Done.

**Comment by nikitamarysolomanpvt** [comment]
> @mschanteltc .

As it is hi-fi i assigned to chantel no specific doubts. Unassigned  @mschanteltc

**Comment by nikitamarysolomanpvt** [comment]
> Screenshot_1578042366
> 
> Submit button should be visible above keyboard, when keyboard is displayed.
> `DONE` / `TICK` in keyboard should submit the pin automatically.
> 
> Also, we need to add test cases too, like:
> 
> 1. Submit button is displayed by default
> 2. Submit button is active only when 5 characters are entered.

@rt4914 I'am facing some issue in the BindingAdapter.
I will create click listener of key-board would be done in a separate PR with new test cases as a part of Low-fi

**Comment by rt4914** [comment]
> > Screenshot_1578042366
> > Submit button should be visible above keyboard, when keyboard is displayed.
> > `DONE` / `TICK` in keyboard should submit the pin automatically.
> > Also, we need to add test cases too, like:
> > 
> > 1. Submit button is displayed by default
> > 2. Submit button is active only when 5 characters are entered.
> 
> @rt4914 I'am facing some issue in the BindingAdapter.
> I will create click listener of key-board would be done in a separate PR with new test cases as a part of Low-fi

@nikitamarysolomanpvt In that case, create an issue to track this.

**Review by rt4914 — APPROVED** [review]

**Pull request merged**